Shared/Samples/Search for web map/SearchForWebMapView.swift

```diff
@@ -0,0 +1,189 @@
+            defer { resultsAreLoading = false }
+            
+            do {
+                try await model.task?.value
+            } catch {
+                self.error = error
+            }
+        }
+        .errorAlert(presentingError: $error)
+    }
+}
+
+private extension SearchForWebMapView {
+    /// A map view that shows an alert and dismisses itself when there is an error loading the map.
+    struct SafeMapView: View {
+        /// The map to show in the map view.
+        let map: Map
+        
+        /// The action to dismiss the view.
+        @Environment(\.dismiss) private var dismiss: DismissAction
+        
+        /// A Boolean value indicating whether the map is being loaded.
+        @State private var mapIsLoading = false
+        
+        /// The error shown in the error alert.
+        @State private var error: Error?
+        
+        var body: some View {
+            ZStack {
+                MapView(map: map)
+                    .task {
+                        mapIsLoading = true
+                        defer { mapIsLoading = false }
+                        
+                        // Show an alert for an error loading the map.
+                        do {
+                            try await map.load()
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                    .errorAlert(presentingError: $error)
+                    .onChange(of: error == nil) { _ in
+                        // Dismiss the view once the error alert is dismissed.
+                        guard error == nil else { return }
+                        dismiss()
+                    }
+                
+                ProgressView()
+                    .opacity(mapIsLoading ? 1 : 0)
+            }
+        }
+    }
+    
+    /// A view that shows a given portal item's info in a row.
+    struct PortalItemRowView: View {
+        /// The portal item to display in the row.
+        let item: PortalItem
+        
+        var body: some View {
+            VStack {
+                HStack {
+                    AsyncImage(url: item.thumbnail?.url) { image in
+                        image
+                            .resizable()
+                            .scaledToFit()
+                            .frame(height: 75)
+                    } placeholder: {
+                        Color(.lightGray)
+                            .frame(width: 110, height: 75)
+                    }
+                    .border(.primary)
+                    .padding([.leading, .top], 10)
+                    
+                    Text(item.title)
+                    
+                    Spacer()
+                }
+                
+                HStack {
+                    Text(item.modificationDate?.formatted(
+                        Date.FormatStyle(date: .abbreviated, time: .omitted)
+                    ) ?? "")
```

Comment:
Maybe…?
```suggestion
                    ) ?? "Last Modified: Unknown")
```

Reply:
`formatted(…)` methods should only be used if there isn't an alternative. In this case, a better way would be to provide the value and format style via a `Text` initializer. That way, SwiftUI is able to efficiently update the view (only when the value and/or style changes, not every time the view is rebuilt).

```swift
if let modificationDate = item.modificationDate {
    Text(modificationDate, format: ...)
} else {
    Text("Unknown")
}
```

Reply:
I used `formatted(…)` because I was trying avoid conditionally showing views. Does that not apply in this case

Reply:
Why avoid that? It shouldn't be done unnecessarily, but this is a case where it makes a lot of sense to do so.